Assets/LeapMotionVR/Scripts/LeapCameraAlignment.cs

```diff
@@ -0,0 +1,556 @@
+      return;
+    }*/
+
+    hasCameras = VRCameras.NONE;
+    if (centerCamera != null) {
+      Camera center = centerCamera.GetComponent<Camera>();
+      if (center != null && center.isActiveAndEnabled) {
+        hasCameras = VRCameras.CENTER;
+      }
+    }
+    if (hasCameras == VRCameras.NONE) {
+      Camera left = leftCamera.GetComponent<Camera>();
+      Camera right = rightCamera.GetComponent<Camera>();
+      if (left != null && left.isActiveAndEnabled &&
+          right != null && right.isActiveAndEnabled) {
+        hasCameras = VRCameras.LEFT_RIGHT;
+      }
+    }
+    if (hasCameras == VRCameras.NONE) {
+      Debug.LogWarning ("Either a central Camera for both eyes, or a Left and Right cameras must be referenced -> enabled = false");
+      enabled = false;
+      return;
+    }
+
+    if (transform.parent == null) {
+      Debug.LogWarning ("Alignment requires a parent object to define the location of the player in the world. enabled -> false");
+      enabled = false;
+      return;
+    }
+
+    if (transform != leftCamera.parent ||
+        transform != centerCamera.parent ||
+        transform != rightCamera.parent) {
+      Debug.LogWarning ("LeapCameraAlignment must be a component of the parent of the camera tranasforms -> enabled = false");
+      enabled = false;
+      return;
+    }
+
+    deviceInfo = (overrideDeviceType) ? new LeapDeviceInfo(overrideDeviceTypeWith) : handController.GetDeviceInfo ();
+    if (deviceInfo.type == LeapDeviceType.Invalid) {
+      Debug.LogWarning ("Invalid Leap Device -> enabled = false");
+      enabled = false;
+      return;
+    }
+
+    if (VRDevice.isPresent &&
+        VRSettings.loadedDevice == VRDeviceType.Oculus) {
+      eyeAlignment = new UserEyeAlignment() {
+        use = true,
+        ipd = OVRPlugin.ipd,
+        eyeDepth = OVRPlugin.eyeDepth,
+        eyeHeight = OVRPlugin.eyeHeight
+      };
+      Debug.Log ("Unity VR Support with Oculus");
+    } else {
+      eyeAlignment = new UserEyeAlignment() {
+        use = false,
+        ipd = 0f,
+        eyeDepth = 0f,
+        eyeHeight = 0f
+      };
+      Debug.Log ("Two-camera stereoscopic alignment");
+    }
+  }
+
+  void Update() {
+    disallowPeripheralTimewarp();
+
+    if (_allowManualTimeAlignment) {
+      if (unlockHold == KeyCode.None ||
+          Input.GetKey(unlockHold)) {
+        // Manual Time Alignment
+        if (Input.GetKeyDown(moreRewind)) {
+          rewindAdjust += 0.1f;
+        }
+        if (Input.GetKeyDown(lessRewind)) {
+          rewindAdjust -= 0.1f;
+        }
+      }
+    }
+    
+    if (Input.GetKeyDown (recenter)) {
+      InputTracking.Recenter();
+    }
+  }
+
+  /// <summary>
+  /// Temporary solution until timecodes on peripheral is fixed.
+  /// </summary>
+  private void disallowPeripheralTimewarp() {
+    DeviceList devices = handController.GetLeapController().Devices;
+    if (devices.Count > 0 && devices[0].Type == Device.DeviceType.TYPE_PERIPHERAL) {
```

Comment:
It makes me nervous that we have this timewarp feature, and we are just turning it off completely for all of our developers.  Maybe we could still leave it on but enter some sort of conservative timewarping mode where we apply a minimal amount of correction instead of full correction to try to help the warping effect?